[clangd] Add Config.Hover.MacroContentsLimit

JaagupAverin opened the pull request:
Currently macro expansions are hard capped at 2048. ~This PR adds the CLI option `--limit-hover-contents` which is passed down in place of the hard coded value.~ This PR adds the` Config.Hover.MacroContentsLimit` config for overriding the default.
Resolves #153355.

Documentation: https://github.com/llvm/clangd-www/pull/136

HighCommander4 commented:
Some high-level feedback:

 1. Hovers contain a variety of types of content (types, values, definitions, initializers, documentation, etc.). This limit applies specifically to macro expansions; therefore, it should have a more specific name, e.g. "limit hover macro expansion" or similar.
 2. Kadir mentioned making this a config option in [this comment](https://github.com/llvm/llvm-project/issues/153355#issuecomment-3187831592), but the current patch is making it a command-line option, is that intentional?

JaagupAverin committed:
[clangd] Add Config.Hover.MacroContentsLimit.

This option allows to override the default limit of 2048 characters when hovering a macro.

JaagupAverin commented:
Hey. Thanks for the response and patience.
Both valid points. I looked at the config options last week and then forgot about them when I got to making the PR and only had CLI options in my mind.
Force pushed a better implementation.

JaagupAverin committed:
[clangd] Config.h formatting.

HighCommander4 reviewed (approved):
Looks good to me, thanks!
